scRNA-seq-advanced/setup/integration-setup.Rmd

```diff
@@ -0,0 +1,533 @@
+  ggplot2::ggtitle("After integration with fastmnn")
+
+```
+
+
+
+
+## Integration, again!
+
+There are more integration tools, like `harmony` for example! 
+`harmony` differs from `fastMNN` (the entire algorithm-aside) in a few key ways:
+
+- `harmony` allows you to specify additional covariates (technology, patient, diagnosis, etc.)
+  - This requires slightly more setup - you'd need to pass in a data frame of covariates, not just a batch vector
+- `harmony` does not calculate corrected expression values and only returns a matrix of corrected PCs
+- a bonus key way is that `harmony` is on CRAN but `batchelor` is on bioconductor.
+
+```{r}
+
+# harmony can either start with PCs or with normalized (uncorrected) expression
+# we already have PCs so let's just ask harmony to use those:
+harmony_result <- harmony::HarmonyMatrix(
+      # PCs
+      data_mat  = reducedDim(combined_sce, "combined_PCA"), 
+      # batch information
+      meta_data = colData(combined_sce)$sample,
+      # don't run PCA since we gave PCs in `data_mat` argument
+      do_pca = FALSE
+    )
+
+# If we wanted to start with expression and have harmony calculate PCs for us:
+#harmony::HarmonyMatrix(
+#  data_mat  = logcounts(combined_sce), 
+#  meta_data = colData(combined_sce)$batch
+#)
+
+# The result is just a bunch of PCs!
+head(harmony_result)
+
+# Let's store this directly into our `combined_sce` object, making sure to name it well!
+reducedDim(combined_sce, "harmony_PCA") <- harmony_result
+
+# Calculate UMAP with the right name for PCs:
+# We do NOT want to overwrite the uncorrected UMAP! Make sure to name this differently with `name`
+combined_sce <- scater::runUMAP(combined_sce, dimred = "harmony_PCA", name = "harmony_UMAP")
+```
+
+#### The before/after reveal, again!
+
+
+```{r}
+# Before
+scater::plotReducedDim(combined_sce,
+                       dimred = "combined_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.4) +
+  ggplot2::ggtitle("Before integration")
+
+
+# After
+scater::plotReducedDim(combined_sce,
+                       dimred = "harmony_UMAP",
+                       colour_by = "sample",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.4) +
+  ggplot2::ggtitle("After integration with harmony")
+
+```
+
+
+## Comparing back to cell types
+
+In an ideal world, we'd have cell types to refer to in UMAPs as well; well-integrated datasets should show more clustering of cell types and less clustering of batches.
+Surprise, it's an ideal world!
+
+Our SCE objects in fact contain this. 
+What are the cell types that have been identified?
+
+```{r}
+unique(combined_sce$celltype)
+```
+
+Let's compare before, fastmnn, and harmony UMAPs highlighting cell types instead of libraries.
```

Comment:
For the purpose of these plots, I wonder if things are a bit clearer if we collapse `Tumor_Myoblast-A`, `Tumor_Myoblast-B` and `Tumor_Myoblast-C` all into `Tumor_Myoblast`. With so many cell types there are a lot of colors here, and I don't think we need to get into the knitty gritty of sub types of cells. I just am curious how that affects the integration picture.

Reply:
Cool cool, glad I posted https://github.com/AlexsLemonade/training-modules/pull/579#discussion_r1036490808 before reading this ;)